src/app/shared/table/table-icon.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+interface IconsMapType {
+  [key: string]: string;
+}
+
+const ICONS_MAP: IconsMapType = {
+  PENDING: 'pending_actions',
+  DONE: 'done',
+  DISCARDED: 'cancel_presentation',
+};
```

Comment:
the interfaces, models should be placed in another file ... we can talk about this in hour angular call from today .... also, the types should be places in a constant file